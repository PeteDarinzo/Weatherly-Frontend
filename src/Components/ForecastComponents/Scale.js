import React from "react";
import Container from '@mui/material/Container';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import { makeStyles } from "@mui/styles";
import { Typography } from "@mui/material";


const useStyles = makeStyles({
  box: {
    border: "2px solid white",
    width: "25px",
    height: "25px",
    // backgroundColor: "black"
  }
});

const Scale = ({ compatibility }) => {

  const classes = useStyles();
  return (
    <Container>
      <Box
        className={classes.box}
        sx={{ backgroundColor: (compatibility > 2 ? "#4caf50" : "#262626") }}
      >
      </Box>
      <Box
        className={classes.box}
        sx={{ backgroundColor: (compatibility > 1 ? "#81c784" : "#262626") }}
      >
      </Box>
      <Box
        className={classes.box} sx={{ backgroundColor: (compatibility > 0 ? "#a5d6a7" : "#262626") }}>
      </Box>
    </Container >
  );
}

export default Scale;