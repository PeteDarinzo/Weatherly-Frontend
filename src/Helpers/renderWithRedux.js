import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from "redux";
import { Provider } from 'react-redux';
import rootReducer from "../Reducers/rootReducer";
import thunk from "redux-thunk"


export default function renderWithRedux(component, { store = createStore(rootReducer, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        {component}
      </Provider>)
  }
}