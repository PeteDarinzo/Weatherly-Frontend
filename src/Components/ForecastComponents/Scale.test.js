import Scale from "./Scale";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import renderWithRedux from "../../Helpers/renderWithRedux";

it("renders scale without crashing", () => {
  render(
    <MemoryRouter>
      <Scale compatibility={2} />
    </MemoryRouter>
  );
});


it("matches scale snapshot", () => {
  const { asFragment } = renderWithRedux(
    <MemoryRouter>
      <Scale compatibility={2} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
