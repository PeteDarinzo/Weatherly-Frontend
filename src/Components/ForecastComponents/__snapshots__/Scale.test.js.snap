// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches scale snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="makeStyles-box-2 MuiBox-root css-165t1cv"
    />
    <div
      class="makeStyles-box-2 MuiBox-root css-rgze7w"
    />
    <div
      class="makeStyles-box-2 MuiBox-root css-1u9s4zi"
    />
  </div>
</DocumentFragment>
`;
